main.js

```diff
@@ -1,74 +1,113 @@
-const addButton = document.querySelector('#add-button');
+/* eslint-disable max-classes-per-file */
 
 const form = document.querySelector('.input-form');
-const main = document.querySelector('.main');
-
-// create Book */
-const createBook = (book) => {
-  const bookContainer = document.createElement('div');
-  bookContainer.className = 'single-book';
-  const bookTitle = document.createElement('h3');
-  bookTitle.className = 'title';
-  const bookAuthor = document.createElement('h3');
-  bookAuthor.className = 'author';
-  const removeButton = document.createElement('button');
-  removeButton.className = 'remove-button';
-  removeButton.textContent = 'Remove';
-  bookTitle.textContent = book.title;
-  bookAuthor.textContent = book.author;
-  bookContainer.append(bookTitle, bookAuthor, removeButton);
-
-  main.insertBefore(bookContainer, form);
-};
-
-/* Add book to localStorage */
-const addBook = () => {
-  let existingBooks = JSON.parse(window.localStorage.getItem('booksStored'));
-  if (existingBooks === null) existingBooks = [];
-  const title = form.elements.title.value;
-  const author = form.elements.author.value;
-  const book = {
-    title,
-    author,
-  };
-  existingBooks.push(book);
-  window.localStorage.setItem('booksStored', JSON.stringify(existingBooks));
-  createBook(book);
-};
-
-const populateBooks = () => {
-  const books = JSON.parse(window.localStorage.getItem('booksStored'));
-  if (localStorage.getItem('booksStored')) {
+
+class Book {
+  constructor(title, author) {
+    this.title = title;
+    this.author = author;
+  }
+}
+
+class Store {
+  getBooks = () => {
+    let books;
+    if (localStorage.getItem('booksStored') === null) {
+      books = [];
+    } else {
+      books = JSON.parse(window.localStorage.getItem('booksStored'));
+    }
+    return books;
+  }
+
+  addBook = (book) => {
+    const books = this.getBooks();
+
+    books.push(book);
+    window.localStorage.setItem('booksStored', JSON.stringify(books));
+  }
+
+  removeBook = (button) => {
+    const books = this.getBooks();
+    const parentDiv = button.parentNode;
+    const myTitle = parentDiv.querySelector('.title').textContent;
+    const myAuthor = parentDiv.querySelector('.author').textContent;
+    const booksLeft = books.filter((book) => book.title !== myTitle && book.author !== myAuthor);
+
+    window.localStorage.setItem('booksStored', JSON.stringify(booksLeft));
+  }
+}
```

Comment:
- [ ] if the user has a list of books when they remove one book and reload the application all the books in the list disappear. please fix this bug before requesting for another review.

Reply:
Please I can't seem to see the bug you are referring to because this does not happen in my code when I remove a book.
If you could direct me to where you think the bug is or the actual error that you are referring to, I would be most grateful. Thank you.

Reply:
I crossed-checked but I can't seem to see the bug you mentioned.

Reply:
As discussed on slack, good job and congratulations on finishing another project/milestone 👏